compiler/hash-parser/src/gen.rs

```diff
@@ -278,151 +412,821 @@ where
+                    FunctionCallExpr {
+                        subject: self.node(Expression::new(ExpressionKind::Variable(
+                            VariableExpr {
+                                name: self.make_access_name_from_str(name, self.current_location()),
+                                type_args: row![&self.wall],
+                            },
+                        ))),
+                        args: self.node(FunctionCallArgs {
+                            entries: row![&self.wall;
+                                self.transform_expr_into_ref(lhs, assigning),
+                                self.make_single_lambda(rhs),
+                            ],
+                        }),
+                    },
+                )))
+            }
+            OperatorFn::Compound { name, assigning } => {
+                // for compound functions that include ordering, we essentially transpile
+                // into a match block that checks the result of the 'ord' fn call to the
+                // 'Ord' enum variants. This also happens for operators such as '>=' which
+                // essentially means that we have to check if the result of 'ord()' is either
+                // 'Eq' or 'Gt'.
+                self.transform_compound_ord_fn(name, assigning, lhs, rhs)
+            }
+        }
+    }
+
+    fn transform_compound_ord_fn(
+        &self,
+        fn_ty: CompoundFn,
+        assigning: bool,
+        lhs: AstNode<'c, Expression<'c>>,
+        rhs: AstNode<'c, Expression<'c>>,
+    ) -> AstNode<'c, Expression<'c>> {
+        let location = lhs.location().join(rhs.location());
+
+        // we need to transform the lhs into a reference if the type of function is 'assigning'
+        let lhs = self.transform_expr_into_ref(lhs, assigning);
+
+        let fn_call = self.node(Expression::new(ExpressionKind::FunctionCall(
+            FunctionCallExpr {
+                subject: self.node(Expression::new(ExpressionKind::Variable(VariableExpr {
+                    name: self.make_access_name_from_str("ord", location),
+                    type_args: row![&self.wall],
+                }))),
+                args: self.node(FunctionCallArgs {
+                    entries: row![&self.wall; lhs, rhs],
+                }),
+            },
+        )));
+
+        // each tuple bool variant represents a branch the match statement
+        // should return 'true' on, and all the rest will return false...
+        // the order is (Lt, Eq, Gt)
+        let mut branches = match fn_ty {
+            CompoundFn::Leq => {
+                row![&self.wall; self.node(MatchCase {
+                    pattern: self.node(Pattern::Or(OrPattern {
+                        variants: row![&self.wall;
+                            self.node(Pattern::Enum(EnumPattern {
+                                name: self.make_access_name_from_str("Lt", location),
+                                args: row![&self.wall],
+                            })),
```

Comment:
Could possibly factor this into a `make_...` function because it is repeated a lot in this function.